### DISCUSSION TITLE
Argo workflow completion notification via Redis

### DISCUSSION THREAD

**Discussion by msidana** [argoproj/argo-workflows#8017] [Q&A]
I have a use case where I am triggering argo workflow from a python application. However, I need a mechanism from argo workflow that it should notify my python application when the workflow execution is completed ( I dont want to do polling). I am already using a pub sub mechansim in my python application. So I want my python app to subscribe to a redis queue and take action once workflow publishes a message on this Redis queue informing its completion.

**Comment by sarabala1979** [comment]
You can configure the exithandle in workflow level which will execute workflow completion. In v3.3 there is new features lifecycle hooks which condition based trigger which will trigger the action whenever first time condition satisfy
